src/NUnitFramework/framework/Constraints/DictionaryContainsKeyConstraint.cs

```diff
@@ -240,17 +242,16 @@ private static MethodInfo FindContainsKeyMethod(Type type)
                 {
                     method = definition
                              .GetMethods(BindingFlags.Instance | BindingFlags.Public)
-                             .FirstOrDefault(m =>
-                                 m.ReturnType == typeof(bool)
-                                 && m.Name == "Contains"
-                                 && !m.IsGenericMethod
-                                 && m.GetParameters().Length == 1
-                                 && m.GetParameters()[0].ParameterType == tKeyGenericArg);
+                             .FirstOrDefault(m => m.ReturnType == typeof(bool) &&
+                                                  m.Name == ContainsMethodName &&
+                                                  !m.IsGenericMethod &&
+                                                  m.GetParameters().Length == 1 &&
+                                                  m.GetParameters()[0].ParameterType == tKeyGenericArg);
 
                     if (method != null)
                     {
 #if NETSTANDARD1_4
-                        method = methods.Single(m => Matched(method, m));
+                        method = methods.Where(x => x.Name == ContainsMethodName).Single(m => MatchedContainsMethod(m, method));
```

Comment:
@jnm2  - Sorry to tag on this...

> (I usually change the TargetFrameworks element so that Test Explorer uses .NET Core to debug.)

I've tried this in the csproj files, I've tried everything over the past few days and I just can't get it to work. I can only get it to target .NetFrameworks... Driving me nuts now.

Reply:
The only csproj to change would be `nunit.framework.tests.csproj`. You'd do `<TargetFrameworks>netcoreapp1.0</TargetFrameworks>`.

Reply:
The only csproj to change would be `nunit.framework.tests.csproj`. You'd do `<TargetFrameworks>netcoreapp1.1</TargetFrameworks>`.

Reply:
@jnm2 🤣 ...

Sorry, I should have said... The `Where()` is more efficient. If I put a `TestContext.WriteLine($"methodInfo");` in the `MatchContainsMethod` with the `Where` it prints out twice, two versions of `Contains`. If I then remove the `Where` it prints out all methods, this is anywhere between 5 and 20 depending on the `Dictionary` type. 

So I left it there, I should add I only know this as when I was debugging, it made it easier to do it with the `Where`, otherwise I would have assumed you were correct.

So I would like to leave it unless I'm missing something and the `ConsoleWrite` statement is overriding the short circuit, but I don't think the compiler is that clever.

Reply:
The first line in `MatchContainsMethods` is `return methodInfo.Name == compareToMethodInfo.Name &&` which is identical to this code:

```c#
if (methodInfo.Name != compareToMethodInfo.Name) return false;

// Console.WriteLine here should only stop at methods named 'Contains'

return ...
```

Reply:
`.Where(x => cond1).Single(x => cond2)` is a less performant but logically equivalent version of `.Single(x => cond1 && cond2)`. 

```c#
method = methods.Single(m => m.Name == ContainsMethodName && MatchedContainsMethod(m, method));
```

So if you inline the `MatchContainsMethods` method into the lambda, it shows what the code is effectively doing:

```c#
method = methods.Single(m =>
    m.Name == ContainsMethodName &&
    m.Name == method.Name &&
    m.GetParameters().Length == 1 &&
    method.GetParameters().Length == 1 &&
    m.GetParameters()[0].Name == method.GetParameters()[0].Name);
```

It's the `m.Name == ContainsMethodName && m.Name == method.Name` that I noticed. :D